theano/sandbox/gpuarray/tests/test_opt.py

```diff
@@ -186,6 +187,25 @@ def test_print_op():
     f(numpy.random.random((5, 5)).astype('float32'))
 
 
+def test_pdbbreakpoint_op():
+    """ Test that PdbBreakpoint ops don't block gpu optimization"""
+    b = tensor.fmatrix()
+
+    # Create a function composed of a breakpoint followed by
+    # some computation
+    condition = tensor.gt(b.sum(), 0)
+    b_monitored = PdbBreakpoint(name='TestBreakpoint')(condition, b)
+    output = b_monitored ** 2
+
+    f = theano.function([b], output, mode=mode_with_gpu)
+
+    # Ensure that, in the compiled function, the computation following the
+    # breakpoint has been moved to the gpu.
+    topo = f.maker.fgraph.toposort()
+    assert isinstance(topo[-2].op, GpuElemwise)
+    assert topo[-1].op == host_from_gpu
```

Comment:
This do not check that the input and output of the break point are on the GPU. If you disable that opt to move to the GPU, I think this tests still pass.